spec/unit/memoizable/memory/delete_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+describe Memoizable::Memory, '#delete' do
+  subject { described_class.new(foo: 1) }
+
+  it 'removes a specific value' do
+    expect(subject.delete(:foo)).to be_nil
+    expect { subject[:foo] }.to raise_error(NameError)
+  end
+end
```

Comment:
__suggestion: (nitpick)__ There should probably be a test that verifies the return value.

Do you know if the new code passes with:

```bash
bundle exec yardstick lib \
  && bundle exec rspec \
  && bundle exec mutant run
```

Reply:
I was covering the return value with:
```ruby
expect(subject.delete(:foo)).to be_nil
```

Based on your suggestion, I split that out into a separate spec. 

Mutant was failing, but that's fixed now. I didn't follow the existing mocked events pattern, because I'm not totally sure I understand what's going on there. Please let me know if I should dig into that more deeply.